src/mapblock_mesh.cpp

```diff
@@ -399,6 +399,31 @@ static void getNodeVertexDirs(v3s16 dir, v3s16 *vertex_dirs)
 	}
 }
 
+static void getNodeTextureCoords(v3f base, const v3f &scale, v3s16 dir, float *u, float *v)
+{
+	if (dir.X > 0 || dir.Y > 0 || dir.Z < 0)
```

Comment:
how come that it's < 0 for z?

Reply:
Dunno. But that must be correct, texture is tiled wrongly otherwise IIRC.